### PR TITLE
Add test on GoogleCloudExtensionPackage.

### PR DESCRIPTION
**Pull request by ILMTitan**

### CODE REVIEW

File: GoogleCloudExtension/GoogleCloudExtension/GoogleCloudExtension.csproj [hunk 1]
```diff
@@ -551,15 +551,15 @@
     <Resource Include="CloudExplorerSources\Gce\Resources\instance_icon_stoped.png" />
     <Resource Include="CloudExplorerSources\Gce\Resources\instance_icon_transition.png" />
     <Content Include="License.txt">
-      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
```
Comment (ivannaranjo) [inline GoogleCloudExtension/GoogleCloudExtension/GoogleCloudExtension.csproj]: I'm sorry, but does it really matter? Can we just copy the files to ensure we always have what comes from the sources?

File: GoogleCloudExtension/GoogleCloudExtension/Properties/AssemblyInfo.cs [hunk 2]
```diff
@@ -34,5 +35,5 @@
 
 // This version number matches the version in the .vsixmanifest. Please update both versions at the
 // same time.
-[assembly: AssemblyVersion("1.2.3.0")]
+[assembly: AssemblyVersion(GoogleCloudExtensionPackage.AssemblyVersion)]
```
Comment (ivannaranjo) [inline GoogleCloudExtension/GoogleCloudExtension/Properties/AssemblyInfo.cs]: Why is this change better now?

File: GoogleCloudExtension/GoogleCloudExtension/GoogleCloudExtensionPackage.cs [hunk 3]
```diff
@@ -73,6 +73,10 @@ public sealed class GoogleCloudExtensionPackage : Package
         /// </summary>
         public const string PackageGuidString = "3784fd98-7fcc-40fc-be3b-b68334735af2";
 
+        // This version number matches the version in the .vsixmanifest. Please update both versions at the
+        // same time.
+        public const string AssemblyVersion = "1.2.3.0";
```
Comment (ivannaranjo) [inline GoogleCloudExtension/GoogleCloudExtension/GoogleCloudExtensionPackage.cs]: This is a fundamental change in how the versioning of the extension is done, why is this change necessary?

File: GoogleCloudExtension/GoogleCloudExtension/GoogleCloudExtension.csproj (hunk 1 above)
Comment (ILMTitan) [inline GoogleCloudExtension/GoogleCloudExtension/GoogleCloudExtension.csproj]: These **Copy Always** items cause the *GoogleCloudExtension* project to have to rebuild every time you want to manually run a unit test, costing just a few seconds. They can add up when making small changes to unit tests, however.

File: GoogleCloudExtension/GoogleCloudExtension/GoogleCloudExtensionPackage.cs (hunk 3 above)
Comment (ILMTitan) [inline GoogleCloudExtension/GoogleCloudExtension/GoogleCloudExtensionPackage.cs]: I wanted to make this a string constant so I could use the same value in both *AssemblyInfo.cs* and *GoogleCloudExtensionPackageTests.cs*. Otherwise, we will need to update this value in three places rather than two.

File: GoogleCloudExtension/GoogleCloudExtension/Properties/AssemblyInfo.cs (hunk 2 above)
Comment (ILMTitan) [inline GoogleCloudExtension/GoogleCloudExtension/Properties/AssemblyInfo.cs]: See my comment in GoogleCloudExtensionPackage.cs.

File: GoogleCloudExtension/GoogleCloudExtension/GoogleCloudExtensionPackage.cs (hunk 3 above)
Comment (ILMTitan) [inline GoogleCloudExtension/GoogleCloudExtension/GoogleCloudExtensionPackage.cs]: I have a change to revert it ready to go if you still have objections.

File: GoogleCloudExtension/GoogleCloudExtensionUnitTests/GoogleCloudExtensionPackageTests.cs [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+        [TestMethod]
+        public void TestPackageValues()
+        {
+            string mockedVersion = "MockVsVersion";
+            string mockedEdition = "MockedEdition";
+            InitPackageMock(
+                dteMock =>
+                {
+
```
Comment (Deren-Liao) [inline GoogleCloudExtension/GoogleCloudExtensionUnitTests/GoogleCloudExtensionPackageTests.cs]: Empty line

Comment (ILMTitan) [inline GoogleCloudExtension/GoogleCloudExtensionUnitTests/GoogleCloudExtensionPackageTests.cs]: Done.

File: GoogleCloudExtension/GoogleCloudExtensionUnitTests/GoogleCloudExtensionPackageTests.cs [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+    /// Tests for the GoogleCloudExtensionPackage class.
+    /// </summary>
+    [TestClass]
+    [DeploymentItem(VsixManifestFileName)]
+    public class GoogleCloudExtensionPackageTests
+    {
+        private const string ExpectedAssemblyName = "google-cloud-visualstudio";
+        private const string VsixManifestFileName = "source.extension.vsixmanifest";
+
+        private static Guid _iidIUnknown = (Guid)Assembly.GetAssembly(typeof(VSConstants))
```
Comment (Deren-Liao) [inline GoogleCloudExtension/GoogleCloudExtensionUnitTests/GoogleCloudExtensionPackageTests.cs]: s_iidIUnknown

Comment (ILMTitan) [inline GoogleCloudExtension/GoogleCloudExtensionUnitTests/GoogleCloudExtensionPackageTests.cs]: Done.

File: GoogleCloudExtension/GoogleCloudExtension/GoogleCloudExtensionPackage.cs (hunk 3 above)
Comment (ILMTitan) [inline GoogleCloudExtension/GoogleCloudExtension/GoogleCloudExtensionPackage.cs]: Changed this. Made the unit test check that the `GoogleCloudExtensionPackage.ApplicationVersion` (which comes from the assembly version) matches the VsixManifest version.

**Comment by ILMTitan** [comment]
PTAL.

File: GoogleCloudExtension/GoogleCloudExtensionUnitTests/GoogleCloudExtensionPackageTests.cs [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+            var serviceProviderMock = new Mock<IServiceProvider>();
+            var dteMock = new Mock<DTE>();
+            var activityLogMock = new Mock<IVsActivityLog>();
+            activityLogMock.Setup(al => al.LogEntry(It.IsAny<uint>(), It.IsAny<string>(), It.IsAny<string>()))
+                .Returns(VSConstants.S_OK);
+            dteSetupAction(dteMock);
+            SetupService<DTE, DTE>(serviceProviderMock, dteMock);
+            SetupService<SVsActivityLog, IVsActivityLog>(serviceProviderMock, activityLogMock);
+            ServiceProvider.CreateFromSetSite(serviceProviderMock.Object);
+            ((IVsPackage)new GoogleCloudExtensionPackage()).SetSite(serviceProviderMock.Object);
```
Comment (Deren-Liao) [inline GoogleCloudExtension/GoogleCloudExtensionUnitTests/GoogleCloudExtensionPackageTests.cs]: I am very surprised, my understanding,  this overrides DTE, service provider etc?
